.github/workflows/ci.yml

```diff
@@ -294,6 +294,42 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+  pr-release:
+    if: ${{ ! startsWith(github.ref, 'refs/tags/') && ! needs.set-nightly.outputs.nightly }}
```

Comment:
Will this not also trigger on directly pushes to branches? It seems like it would be best to simply check that `github.event_name == 'pull_request'`,

Reply:
Hmm, but then that will only trigger when the PR is opened or its metadata modified in some way: just pushing to the branch would not trigger updating the release.

Reply:
@semorrison That is not my understanding. From Github's documentation on the [`pull_request`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) event, all forms of pull request modifications (including pushing to branch, which falls under `synchronize`) have `pull_request` event name.

Reply:
Instead I've added an additional check that the `github.event.number` is nonempty.

Reply:
Ah, I hate the Github docs. That page gives no definition for `synchronize`!

I'll try your suggestion.